From db97fe41f90da2fe8df636b28715d301035ab3a6 Mon Sep 17 00:00:00 2001
From: Vu Dang <vu.dang.te@renesas.com>
Date: Thu, 17 Jul 2025 15:29:53 +0700
Subject: [PATCH 1/2] arm64: dts: renesas: enable hardware video codec

This commit enables hardware video IPs for RZG2L-EVK and RZV2L-EVK

Upstream-Status: Pending

Signed-off-by: Vu Dang <vu.dang.te@renesas.com>
---
 arch/arm64/boot/dts/renesas/r9a07g054.dtsi    | 40 +++++++++++++++++++
 .../boot/dts/renesas/r9a07g054l2-smarc.dts    | 24 +++++++++++
 .../boot/dts/renesas/rzg2l-smarc-som.dtsi     | 33 +++++++++++++++
 3 files changed, 97 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r9a07g054.dtsi b/arch/arm64/boot/dts/renesas/r9a07g054.dtsi
index 66894b676c01..ca6d5c7b4b7b 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g054.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a07g054.dtsi
@@ -536,6 +536,46 @@ channel1 {
 			};
 		};
 
+		fcpcs: vcp4@10820000 {
+			compatible = "renesas,vcp4-fcpcs";
+			reg = <0 0x10820000 0 0x200>;
+			renesas,#ch = <0>;
+		};
+
+		vcpl4: vcp4@10800000 {
+			compatible = "renesas,vcp4-vcpl4";
+			reg = <0 0x10800000 0 0x200>, <0 0x10810200 0 0x200>;
+			interrupts = <GIC_SPI 161 IRQ_TYPE_LEVEL_HIGH>,
+					<GIC_SPI 162 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G054_H264_CLK_A>,
+				<&cpg CPG_MOD R9A07G054_H264_CLK_P>;
+			power-domains = <&cpg>;
+			resets = <&cpg R9A07G054_H264_X_RESET_VCP>,
+				<&cpg R9A07G054_H264_CP_PRESET_P>;
+			renesas,#ch = <0>;
+			renesas,#fcp_ch = <0>;
+		};
+
+		isu: isum@10840000 {
+			compatible = "renesas,isum";
+			reg = <0 0x10840000 0 0x01F4>;
+			interrupts = <GIC_SPI 157 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 158 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 159 IRQ_TYPE_LEVEL_HIGH>,
+				<GIC_SPI 160 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD R9A07G054_ISU_ACLK>,
+				<&cpg CPG_MOD R9A07G054_ISU_PCLK>;
+			clock-names = "isu_aclk", "isu_pclk";
+			resets = <&cpg R9A07G054_ISU_ARESETN>,
+				<&cpg R9A07G054_ISU_PRESETN>;
+			reset-names = "aresetn", "presetn";
+			renesas,#ch = <0>;
+			renesas,#rpf = <1>;
+			renesas,#wpf = <1>;
+			renesas,has-rs;
+			power-domains = <&cpg>;
+		};
+
 		i2c0: i2c@10058000 {
 			#address-cells = <1>;
 			#size-cells = <0>;
diff --git a/arch/arm64/boot/dts/renesas/r9a07g054l2-smarc.dts b/arch/arm64/boot/dts/renesas/r9a07g054l2-smarc.dts
index b3e6016880dd..e3df6bbe16f5 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g054l2-smarc.dts
+++ b/arch/arm64/boot/dts/renesas/r9a07g054l2-smarc.dts
@@ -35,4 +35,28 @@
 / {
 	model = "Renesas SMARC EVK based on r9a07g054l2";
 	compatible = "renesas,smarc-evk", "renesas,r9a07g054l2", "renesas,r9a07g054";
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		/* device specific region for contiguous allocations */
+		drp_reserved: DRP-AI {
+			reusable;
+			reg = <0x0 0x80000000 0x0 0x20000000>;
+		};
+
+		image_buf0: image_buf@0 {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x0 0xB0000000 0x0 0x04000000>;
+			label = "image_buf0";
+		};
+
+		isp_work0: SIMPLE_ISP {
+			reusable;
+			reg = <0x0 0xB4000000 0x0 0x03000000>;
+		};
+	};
 };
diff --git a/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi b/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
index 4409c47239b9..995a539335bf 100644
--- a/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2l-smarc-som.dtsi
@@ -35,6 +35,39 @@ memory@48000000 {
 		reg = <0x0 0x48000000 0x0 0x78000000>;
 	};
 
+	reserved-memory {
+                #address-cells = <2>;
+                #size-cells = <2>;
+                ranges;
+
+                global_cma: linux,cma@58000000 {
+                        compatible = "shared-dma-pool";
+                        linux,cma-default;
+                        reusable;
+                        reg = <0x0 0x58000000 0x0 0x10000000>;
+                };
+
+                mmp_reserved: linux,multimedia {
+                        compatible = "shared-dma-pool";
+                        reusable;
+                        reg = <0x0 0x68000000 0x0 0x8000000>;
+                };
+
+        };
+
+        mmngr {
+                compatible = "renesas,mmngr";
+                memory-region = <&mmp_reserved>;
+        };
+
+        mmngrbuf {
+                compatible = "renesas,mmngrbuf";
+        };
+
+        vspm_if {
+                compatible = "renesas,vspm_if";
+        };
+
 	reg_1p8v: regulator-1p8v {
 		compatible = "regulator-fixed";
 		regulator-name = "fixed-1.8V";
-- 
2.43.0

